test/helpers/test.go

```diff
@@ -553,26 +608,77 @@ func (t *TestOptions) SetGitHubToken() {
 }
 
 // GetPullRequestWithTitle Returns a pull request with a matching title
-func (t *TestOptions) GetPullRequestWithTitle(client *github.Client, ctx context.Context, repoOwner string, repoName string, title string) (*github.PullRequest, error) {
-	pullRequestList, _, err := client.PullRequests.List(ctx, repoOwner, repoName, nil)
+func (t *TestOptions) GetPullRequestWithTitle(provider gits.GitProvider, repoOwner string, repoName string, title string) (*gits.GitPullRequest, error) {
+	pullRequestList, err := provider.ListOpenPullRequests(repoOwner, repoName)
 	if err != nil {
 		return nil, err
-
 	}
 
-	var matchingPR *github.PullRequest
 	for _, pullRequest := range pullRequestList {
-		if *pullRequest.Title == title {
-			matchingPR = pullRequest
+		if pullRequest.Title == title {
+			return pullRequest, nil
 		}
 	}
 
-	return matchingPR, nil
+	return nil, nil
+}
+
+// ApprovePRFromLogOutput takes the default provider, the approver user's provider, git info, and the output from a command that
+// created a PR, and adds the approver user as a collaborator, accepts the invitation, and approves the PR.
+func (t *TestOptions) ApprovePRFromLogOutput(provider gits.GitProvider, approverProvider gits.GitProvider, gitInfo *gits.GitRepository, output string) {
```

Comment:
Could we please either use `PR` or `PullRequest` in function names. Now we are doing both which imo gets confusing. Personally I prefer `PullRequest` even if it is a bit longer. IMO things like "PRF" get hard to "parse".

Reply:
Fine by me - I’ll scrub the names.